enumerable_methods.rb

```diff
@@ -54,94 +66,147 @@ def my_select
     end
   end
 
-  def my_all?
-    return to_enum unless block_given?
-
+  def my_all?(cls = nil)
     arr = self
-    if arr.is_a? Array
+    if cls.nil?
+      if arr.is_a? Array
+        x = true
+        i = 0
+        while i < arr.length
+          if block_given?
+            x = false unless yield(arr[i])
+          else
+            x = false unless arr[i]
+          end
+          i += 1
+        end
+        x
+      else
+        puts 'Error: Parameter is not an array!'
+      end
+    else
       x = true
       i = 0
+      cls.class == Integer.class ? cls = cls : cls = cls.class
+      puts cls
       while i < arr.length
-        x = false unless yield(arr[i])
+        x = false unless arr[i].class == cls
         i += 1
       end
```

Comment:
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern